src/apps/service/apps/job_listings.clj

```diff
@@ -41,8 +41,8 @@
 (defn- job-supports-sharing?
   [apps-client perms rep-steps {:keys [parent_id id]}]
   (and (nil? parent_id)
-       (job-permissions/job-steps-support-job-sharing? apps-client (rep-steps id))
-       (= (get perms id) "own")))
+       (= (get perms id) "own")
+       (job-permissions/job-steps-support-job-sharing? apps-client (rep-steps id))))
```

Comment:
I switched the order of the conditions here so that the simpler comparison would happen first.